scripts/ci/build-and-test.sh

```diff
@@ -10,6 +10,26 @@ cd $(dirname $0)/../..
 source scripts/ci/sources/mode.sh
 source scripts/ci/sources/tunnel.sh
 
+# Get commit diff
+if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then
+  diff=$(git diff --name-only $TRAVIS_COMMIT_RANGE)
+else
+  diff=$(git diff --name-only $TRAVIS_BRANCH...HEAD)
+fi
+
+# Check if tests can be skipped
+skip_tests=true
+for filename in $diff; do
+  if ! [[ "$filename" =~ .*\.md ]]; then
+    skip_tests=false
+    break
+  fi
+done
```

Comment:
Cool, I just learned something new from you. 

Only two things:
1. ``[[ "$TRAVIS_PULL_REQUEST" = "false" ]]`` would print as ``+[[ false = \f\a\l\s\e ]]`` in Travis output, and quotes are unnecessary.  
[ "$TRAVIS_PULL_REQUEST" = "false" ] would print as ``+[[ false = false ]]. I think it's more readable. 

2.  I changed the regex ``${fileDiff} =~ ^(.*\.md\s*)*$`` to ``${fileDiff} =~ ^(.*\.md\s*)+$``. Although I doubt anyone would push an empty commit...